src/main/java/de/hpi/datastreams/processors/ServerProcessor.java

```diff
@@ -83,23 +85,12 @@ private void initializeModel() {
     private KeyRange getKeyRange() {
         Integer smallestKey = 0;
         Integer largestKey = 0;
-        boolean smallestKeySet = false;
-        boolean largesttKeySet = false;
-
-        KeyValueIterator<Integer, Float> iter = this.weights.all();
-        while (iter.hasNext()) {
-            KeyValue<Integer, Float> entry = iter.next();
-            if(entry.key < smallestKey || !smallestKeySet) {
-                smallestKey = entry.key;
-                smallestKeySet = true;
-            }
-
-            if(entry.key > largestKey || !largesttKeySet) {
-                largestKey = entry.key;
-                largesttKeySet = true;
-            }
+
+        if(!this.weights.get(0L).isEmpty()) {
```

Comment:
We should in general avoid using `KeyValueStore`s with just a single utilized key.
But this more or less refers to the comment above.